sdk/java-sdk/src/main/java/kalix/javasdk/eventsourcedentity/EventSourcedEntity.java

```diff
@@ -179,6 +179,11 @@ protected final Effect.Builder<S> effects() {
 
     interface OnSuccessBuilder<S> {
 
+      /**
+       * Delete the entity. No addition events are allowed.
+       */
+      OnSuccessBuilder<S> deleteEntity();
```

Comment:
@aludwiko I changed this to your suggestion. Much better. https://github.com/lightbend/kalix-jvm-sdk/pull/1361/commits/636dfa605e2df089c1440cff79c20d8d889de21c
```
  effects()
    .emitEvent(finalEvent)
    .deleteEntity()
```

Then we support multiple final events as well.

Reply:
Nice